cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/oidc/OIDCUtil.java

```diff
@@ -171,46 +173,126 @@ private static Key getPrivateKey() {
         return privateKey;
     }
     
-    public static RelyParty getRelyParty(String clientID) {
-        if (relyParties == null) {
-            // add all rely parties
- 
-            log.debug("Reading RelyParties properties from: " + CONFIG);
-            relyParties = new HashMap<String, RelyParty>();
-            PropertiesReader pr = new PropertiesReader(CONFIG);
-            MultiValuedProperties config = pr.getAllProperties();
-            if (config == null || config.keySet() == null)
-            {
-                throw new RuntimeException("failed to read any OIDC property ");
+    private static Set<String> getClientIDs(MultiValuedProperties config) {
+        Set<String> clientIDs = new HashSet<String>();
+        Set<String> keys = config.keySet();;
+        for (String k : keys) {
+            clientIDs.add(k.split("\\.")[0]);
+        }
+        
+        return clientIDs;
+    }
+    
+    private static void checkKey(MultiValuedProperties config, Set<String> keys, String key) {
+        if (!keys.contains(key)) {
+            throw new IllegalStateException("missing key " + key + " in " + CONFIG);
+        } else if (config.getProperty(key).get(0).isEmpty()) {
+            throw new IllegalStateException("missing value for " + key + " in " + CONFIG);
+        }
+    }
+    
+    private static String getSecret(MultiValuedProperties config, Set<String> keys, String clientID) {
+        String secretKey = clientID + ".secret";
+        checkKey(config, keys, secretKey);
+        return config.getProperty(secretKey).get(0);
+    }
+    
+    private static String getDescription(MultiValuedProperties config, Set<String> keys, String clientID) {
+        String descriptionKey = clientID + ".description";
+        checkKey(config, keys, descriptionKey);
+        return config.getProperty(descriptionKey).get(0);
+    }
+    
+    private static boolean getSignDocuments(MultiValuedProperties config, Set<String> keys, String clientID) {
+        String signDocumentsKey = clientID + ".sign-documents";
+        checkKey(config, keys, signDocumentsKey);
+        return Boolean.valueOf(config.getProperty(signDocumentsKey).get(0));
+    }
+    
+    private static List<Claim> getClaims(MultiValuedProperties config, Set<String> keys, String clientID) {
+        String claimsKey = clientID + ".claims";
+        checkKey(config, keys, claimsKey);
+        List<Claim> claims = new ArrayList<Claim>();
+        final String[] claimsArray = config.getProperty(claimsKey).get(0).split(" ");
+        for (String claim : claimsArray) {
```

Comment:
You can just use the Claim.getClaim(string) method to construct the claim... no need for the if statements.

Reply:
Good catch.